sycl/plugins/rocm/pi_rocm.cpp

```diff
@@ -4281,9 +4281,20 @@ pi_result rocm_piextUSMFree(pi_context context, void *ptr) {
     ScopedContext active(context);
     unsigned int type;
     hipPointerAttribute_t hipPointerAttributeType;
-    result =
-        PI_CHECK_ERROR(hipPointerGetAttributes(&hipPointerAttributeType, ptr));
+    hipError_t ret = hipPointerGetAttributes(&hipPointerAttributeType, ptr);
     type = hipPointerAttributeType.memoryType;
+#ifdef __HIP_PLATFORM_NVIDIA__
+    // The NVIDIA hipPointerGetAttributes implementation doesn't know about
```

Comment:
What about other calls to `hipPointerGetAttributes` which will continue to return error?

Reply:
That's a good point, I only ran into this call failing, but yeah it will happen with the others too, I'll update them

Reply:
please consider doing this adjustment in the function itself, if feasible

Reply:
Just a quick update on this patch, I'm currently looking at this and some of the other HIP workarounds in the ROCm plugin and figuring out if I can fix them in the HIP headers directly and submit the fixes to HIP upstream.

However I do think we'll probably still need the workarounds in the ROCm plugin, at least until the next HIP release and if my fixes get approved. So I'll come back to this patch in a bit and make it a proper workaround for all the other uses of `hipPointerGetAttributes` as well.

Reply:
Is it changed up-stream?

Reply:
@npmiller looks like AMD put a patch in for this last week https://github.com/ROCm-Developer-Tools/hipamd/commit/88f1622d20efd18aa3bb05f74bba05ec943fb2c7. You should be able to do a proper fix now.